### PR TITLE
[TE] spawn pinot connections in parallel on initialization

### PR DESCRIPTION
**Pull request by apucher**
* spawn pinot connections in parallel on initialization

* speedup thirdeye startup by 30sec-1min

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/cache/ResultSetGroupCacheLoader.java [hunk 1]
```diff
@@ -102,4 +104,47 @@ private static String format(ResultSetGroup result) {
     }
     return sb.toString();
   }
+
+  private static Connection[] fromHostList(final String[] thirdeyeBrokers) throws Exception {
+    ExecutorService executor = Executors.newCachedThreadPool();
+
+    Collection<Future<Connection>> futures = new ArrayList<>();
+    for(int i=0; i<MAX_CONNECTIONS; i++) {
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/cache/ResultSetGroupCacheLoader.java]: Format this line.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/cache/ResultSetGroupCacheLoader.java [hunk 2]
```diff
@@ -102,4 +104,47 @@ private static String format(ResultSetGroup result) {
     }
     return sb.toString();
   }
+
+  private static Connection[] fromHostList(final String[] thirdeyeBrokers) throws Exception {
+    ExecutorService executor = Executors.newCachedThreadPool();
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/cache/ResultSetGroupCacheLoader.java]: Close the pool at the end of this method. You could use AnomalyUtils.safelyShutdownExecutionService to shut down the pool. We might want to move safelyShutdownExecutionService to some other Utils class if it is used everywhere.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/cache/ResultSetGroupCacheLoader.java [hunk 3]
```diff
@@ -102,4 +104,47 @@ private static String format(ResultSetGroup result) {
...
+          return ConnectionFactory.fromHostList(thirdeyeBrokers);
+        }
+      }));
+    }
+
+    return fromFutures(futures);
+  }
+
+  private static Connection[] fromZookeeper(final PinotThirdEyeDataSourceConfig pinotThirdEyeDataSourceConfig) throws Exception {
+    ExecutorService executor = Executors.newCachedThreadPool();
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/cache/ResultSetGroupCacheLoader.java]: Close this pool at the of this method.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/cache/ResultSetGroupCacheLoader.java [hunk 4]
```diff
@@ -102,4 +104,47 @@ private static String format(ResultSetGroup result) {
...
+
+    Collection<Future<Connection>> futures = new ArrayList<>();
+    for(int i=0; i<MAX_CONNECTIONS; i++) {
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/cache/ResultSetGroupCacheLoader.java]: Format this line.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/cache/ResultSetGroupCacheLoader.java (hunk 1 above)
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/cache/ResultSetGroupCacheLoader.java]: ?

**Pull request merged**